File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        myMap = {}
        
        
        for i, n in enumerate(strs):
            sortedStr = ''.join(sorted(n))
            if (sortedStr not in myMap):
                myMap[sortedStr] = [n]
            else:
                myMap[sortedStr].append(n)
        
        return myMap.values()
            

File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    // [..."apple"].sort().join('') = 'aelpp'
    
    let myObj = strs.reduce((acc, curr) => {
        let currSorted = [...curr].sort().join('')  // sorted string
        if (!acc[currSorted]) acc[currSorted] = []
        acc[currSorted].push(curr)
        return acc
    }, {})
    
    return Object.values(myObj) 
};